src/transactions/transactions.service.ts

```diff
@@ -227,7 +227,7 @@ export class TransactionsService {
       };
 
       if (status === TransactionStatus.Succeeded) {
-        payload.result = 'placeholder'; // TODO @monitz87: use real result when we eliminate TQs
+        payload.result = transaction.receipt;
```

Comment:
This shouldn't be the receipt, but rather the result of running the transaction as resolved by the promise returned by `transaction.run`. The idea is for the subscriber to receive (for example) the id of the venue they just created, as if they had called the endpoint without subscribing.

This could be done with the current setup by assigning the result of `transaction.run` to a variable, then calling `catch` on it and then passing it to `assemblePayload` (which would have to be made async and might also involve some ugly type assertions).

My opinion though is that the `PolymeshTransactionBase` class in the SDK should be modified in one of two ways:

1. Adding a `public result?: TransformedReturnValue` property that is set before returning the result in the `run` method
2. Adding a `private _result?: TransformedReturnValue` property which is set in the same way as 1 and a `public get result(): TransformedReturnValue` getter that throws an error if called before the private variable has been set

I prefer 2 because it avoids clunky type assertions, and you can always avoid the error if you check that the transaction's status is `Succeeded` (by the way, the JSDoc for the getter should explain this)

Reply:
[DA-407](https://polymesh.atlassian.net/browse/DA-407)